src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyFollowedBySemicolonHasLineBreakAfter.java

```diff
@@ -0,0 +1,15 @@
+/*
+RightCurly
+option = ALONE_OR_SINGLELINE
+tokens = LITERAL_TRY , LITERAL_CATCH , LITERAL_FINALLY , LITERAL_IF , LITERAL_ELSE , CLASS_DEF , METHOD_DEF , CTOR_DEF , LITERAL_FOR , LITERAL_WHILE , LITERAL_DO , STATIC_INIT , INSTANCE_INIT , ANNOTATION_DEF , ENUM_DEF , INTERFACE_DEF , RECORD_DEF , COMPACT_CTOR_DEF
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;
+
+public class InputRightCurlyFollowedBySemicolonHasLineBreakAfter {
+
+        Object o = new Object() {
```

Comment:
@shubh220922 please extend test cases, this is what I would expect from issue description:

```
public class Test {
    Object o = new Object() {
    }; int i = 0;  // should be a violation

    class Inner {
    } int x = 0;  // violation

    void m() {
    } int y = 0;  // violation
}

```

Idea is the similar for each field/method above, we already place violations on inner class and method right curly, we should do the same for anonymous class.

Reply:
@nrmancuso DONE.

Modified input file.